src/vs/workbench/contrib/search/browser/searchActions.ts

```diff
@@ -459,6 +463,22 @@ class ReplaceActionRunner {
 		const opInfo = getElementsToOperateOnInfo(this.viewer, element, this.configurationService.getValue<ISearchConfigurationProperties>('search'));
 		const elementsToReplace = opInfo.elements;
 
+		if (elementsToReplace.length === 0) {
+			return;
+		}
+
+		const currentBottomFocusElement = elementsToReplace[elementsToReplace.length - 1];
+		const currentTopFocusElement = elementsToReplace[0];
+
+		if (!(currentBottomFocusElement instanceof Match)) {
+			const nextFocusElement = this.getElementToFocusAfterRemoved(this.viewer, currentBottomFocusElement, currentTopFocusElement);
```

Comment:
moved all instances of getElementToFocusAfterRemoved to before the removal because it needs a an existing element to check against.